nestkernel/sp_manager.cpp

```diff
@@ -227,12 +227,21 @@ SPManager::disconnect_single( index sgid,
     SPBuilder* cb = new SPBuilder( *sources, *targets, *conn_spec, syn );
     cb->change_connected_synaptic_elements(
       sgid, target->get_gid(), target->get_thread(), -1 );
+    const std::string syn_name = ( *syn )[ names::model ];
+    disconnect( sgid,
+      target,
+      target_thread,
+      kernel().model_manager.get_synapsedict()->lookup( syn_name ) );
+  }
+  else
+  {
+    // Disconnect if Structural plasticity is not activated.
+    const std::string syn_name = ( *syn )[ names::model ];
+    disconnect( sgid,
+      target,
+      target_thread,
+      kernel().model_manager.get_synapsedict()->lookup( syn_name ) );
```

Comment:
Yes this is right thanks for noticing, I was moving things around and at the end remained the same code. I am correcting it now.